e2e/libs/recurringjob/crd.py

```
@@ -116,12 +116,15 @@ def wait_for_systembackup_state(self, job_name, expected_state):
+                    if state == expected_state:
+                        return
 
-                return
-            except AssertionError:
-                logging(f"Waiting for systembackup in state '{state}' to reach state '{expected_state}' ({i}) ...")
-                time.sleep(self.retry_interval)
+            except Exception as e:
+                logging(f"Waiting for systembackup created by job {job_name} to reach state {expected_state} failed: {e}")
+
+            time.sleep(self.retry_interval)
+        assert False, logging(f"Waiting for systembackup created by job {job_name} to reach state {expected_state} failed")
```

Ruff (0.8.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)